king_phisher/server/fs_utilities.py

```diff
@@ -83,3 +86,77 @@ def chown(path, user=None, group=constants.AUTOMATIC, recursive=True):
 		iterator = (path,)
 	for path in iterator:
 		os.chown(path, user, group)
+
+def access(path, mode, user=None, group=constants.AUTOMATIC):
+	"""
+	This is a high-level wrapper around :py:func:`os.access` to provide
+	additional functionality. Similar to `os.access` this high-level wrapper
+	will test the given path for a variety of access modes. Additionally however,
+	*user* or *group* can be specified to test against a specific of user or group.
+
+	.. versionadded:: 1.14.0
+
+	:param str path: The path to test access for.
+	:param str mode: The mode to test access for. Set to
+		`R_OK` to test for readability, `W_OK` for writability and
+		`X_OK` to determine if path can be executed.
+	:param user: The user to test permissions for.
+	:type user: int, str, ``None``
+	:param group: The group to test permissions for. If set to
+		:py:class:`~king_phisher.constants.AUTOMATIC`, the group that *user*
+		belongs too will be used.
+	:type group: int, str, ``None``, :py:class:`~king_phisher.constants.AUTOMATIC`
+	:return: Returns ``True`` only if the user or group has the mode of permission specified else returns ``False``.
+	:rtype: bool
+	"""
+	if (user is constants.AUTOMATIC or user is None) and (group is constants.AUTOMATIC or group is None):
+		raise ValueError('either user or group must be specified')
+
+	if isinstance(user, str):
```

Comment:
So instead of handling user and group seperately like you do, you should "resolve" the user and group arguments to numeric uid and gid values like the `chown` function does. I recommend you reuse that code by placing it in a private function (prefix it with `_`).

That way you can consolidate the logic and move on to (in english):
1. Is the owner the UID (resolved from *user*) I'm looking for and does the owner have the permissions (the *mode* argument) I'm looking for?
1. Is the group a GID I'm looking for and does the group have the permissions I'm looking for?
    * GID checks are a bit more tricky, if `group` is `constants.AUTOMATIC`, then you should check **all the groups that the user is a member of**, if `group` is `None` skip checking the group altogether, if `group` is an explicit number then only check that one
1. Check if everyone has the access I'm looking for.